ui/src/shared/components/TimeMachine/TimeMachine.tsx

```diff
@@ -206,21 +214,27 @@ class TimeMachine extends PureComponent<Props, State> {
 
   private get source(): Source {
     const {source, sources, queryDrafts} = this.props
-    const {selectedSource} = this.state
+    const {selectedSource, useDynamicSource} = this.state
 
     if (selectedSource) {
       return selectedSource
     }
 
-    const query = _.get(queryDrafts, 0)
-    const querySource = _.get(query, 'source')
+    // return current source
+    if (useDynamicSource) {
+      return source
+    }
+
+    const queryDraft = _.get(queryDrafts, 0)
+    const querySource = _.get(queryDraft, 'source')
 
-    if (!query || !querySource) {
+    if (!queryDraft || !querySource) {
       return source
     }
 
+    // return source from queryDraft
     const foundSource = sources.find(
-      s => s.links.self === _.get(query, 'source', null)
+      s => s.links.self === _.get(queryDraft, 'source', null)
```

Comment:
I feel like the right side of this equality could be `=== querySource` ?

Reply:
missed that. you're right :)